### PR TITLE
integrate facebooc with angularjs, leafletjs, d3, and timbrejs

### PR DESCRIPTION
**Pull request by Gogistics**
make frontend development more scalable with angularjs, integrate facebooc with leafletjs and d3 for data visualization, integrate facebooc with midi framework timbrejs

### CODE REVIEW

**Comment by jserv** [comment]
Does commit 327b9cb91151133a51cdb8b4d6f7e748fbd60680 only depend on AngularJS?

File: static/css/main.css [hunk 1]
```diff
@@ -1,5 +1,5 @@
 body {
-  padding-top: 20px;
+  /* padding-top: 20px; by alantai */
```
Comment (jserv) [inline static/css/main.css:2]: Don't annotate your name here since we have version control systems like `Git`, that implies we can always track every change via commands like `git blame`.

File: templates/footer.html [hunk 2]
```diff
@@ -2,7 +2,26 @@
...
-    <script src="//netdna.bootstrapcdn.com/bootstrap/3.1.1/js/bootstrap.min.js"></script>
+    <script type="text/javascript" src="/static/jq_plugin/jquery-1.12.3.min.js"></script>
+    <script type="text/javascript" src="/static/bootstrap-3.3.6-dist/js/bootstrap.min.js"></script>
+    <!-- angularjs -->
+    <script type="text/javascript" src="/static/angular-1.5.5/angular.min.js"></script>
+    <script type="text/javascript" src="/static/angular-1.5.5/angular-animate.min.js"></script>
+    <!-- leaflet -->
+    <script type="text/javascript" src="/static/leaflet-0.7/leaflet.min.js"></script>
+    <!-- d3 -->
+    <script type="text/javascript" src="/static/d3/d3.min.js"></script>
```
Comment (jserv) [inline templates/footer.html:13]: Why did you need d3.js?

File: static/js/main.js [hunk 3]
```diff
@@ -0,0 +1,252 @@
+/**/
+(function($) {
```
Comment (jserv) [inline static/js/main.js:2]: Please split the externals JavaScript frameworks into several functional commits. Also, it looks file `static/js/main.js` was written by you, right? If so, please decouple from this commit because we always want to ease the maintenance of external components.

**Comment by Gogistics** [comment]
1. main.css
   I'll remove it next week. Previously, my intern worked on the integration with this account and I asked him to modify the css as I mentioned so he put my name on it to distinguished what he did.
2. d3
   I wanted my intern to learn how to integrate D3 with Leaflet for advanced data visualization of geo-information. The location mark was implemented with d3 and css,
3. main.js and other external modules
   Yes, I asked my intern to follow my templates writing the script, main.js, to implement the frontend MVC structure with Angular.js I'll work on splitting the externals JavaScript frameworks into several functional commits next week.

**Comment by jserv** [comment]
@Gogistics : Understood.
By the way, we should mention the name or nick of somebody rather than plain word "intern".
Follow "Learning by doing"  as a guide to teaching and learning methods.

**Comment by jserv** [comment]
@Gogistics : ping? Are you still interested in improving Facebooc?

**Comment by Gogistics** [comment]
The intern's nickname is Micky and he leaves soon for summer trips. Besides, two new projects are coming so I don't probably have that much time to contribute without my intern. I'll break down the commits I promised during my spare time this week.

**Comment by jserv** [comment]
Drop this PR in favor of #21
